chapters/chap03-sec03.tex

```diff
@@ -36,25 +36,25 @@ \section{\zeth~statement after primitive instantiation}\label{instantiation:stat
+The cost of $\groth$ zk-SNARK~\cite{groth2016size} proof verification is linear in the number of primary inputs, since each input acts as a scalar in a costly scalar multiplication of a curve point in $\gset_1$. Hence, while packing slightly increases the prover cost -- by adding constraints to the circuit -- it simplifies the verifier's work.
 
-We detail in this section the method by which we encode (resp.~decode) a set of binary strings (resp.~field elements) into a set of field elements (resp.~binary strings). In the rest of this section, the notion of \emph{packing policy} is used to refer to the set of \emph{packing} and \emph{unpacking} functions.
+In this section we detail the method by which we encode (resp.~decode) a set of binary strings to (resp.~from) sets of field elements. In the rest of this section, the notion of \emph{packing policy} refers to the set of \emph{packing} and \emph{unpacking} functions.
 
 The set of primary inputs is composed of the input nullifiers, the output commitments, the public values (see~\cite[Section 3.4.3]{zethpaper}) along with the signature hash and the authentication tags for security (malleability fix, see~\cref{appendix:trnm}). The complete description of the public inputs is represented in~\cref{instantiation:eq:primary-inputs}.
 %
 \begin{equation}\label{instantiation:eq:primary-inputs}
     (\indexedset{\priminputs.\nf{i}}{i \in [\jsin]}, \indexedset{\priminputs.\cms{j}}{j \in [\jsout]}, \vin, \vout, \hsig, \indexedset{\priminputs.\htags{i}}{i \in [\jsin]})
 \end{equation}
 
-Consider the subset of primary inputs that are binary strings, that is the nullifiers $\nfs{}$, the public values $\vin$ and $\vout$, the signature hash $\hsig$ and the authentication tags $\htags{}$.
-For each element $x$ in this subset, $\alpha_x = \ceil {\len{x} / \bnFieldBitCap}$, represent the number of field elements to encode all of the element's bits. Let $\beta_x = \floor{\len{x} / \bnFieldBitCap}$ represent the number of field elements which are fully ``filled'' and let $\gamma_x = \len{x} \pmod{\bnFieldBitCap}$ denote the number of remaining bits. For simplicity, as we assume $\zvalueLen < \bnFieldBitCap$, we only define the notation $\gamma_\notev = \zvalueLen$ for the public values $\vin$ and $\vout$.
+The primary inputs that consist of binary strings are: the nullifiers $\nfs{}$, the public values $\vin$ and $\vout$, the signature hash $\hsig$ and the authentication tags $\htags{}$.
+
+For a binary string $x$, let $\alpha_x = \ceil {\len{x} / \bnFieldBitCap}$ be the number of field elements required to completely encode $x$ and let $\beta_x = \floor{\len{x} / \bnFieldBitCap}$ be the number of field elements whose capacity is fully used. Let $\gamma_x = \len{x} \pmod{\bnFieldBitCap}$ be the number of remaining or ``residual'' bits remaining after fully using $\beta_x$ field elements.
```

Comment:
`$\gamma_x = \len{x} \pmod{\bnFieldBitCap}$ be the number of remaining or ``residual'' bits remaining after fully using $\beta_x$ field elements.` -> `$\gamma_x = \len{x} \pmod{\bnFieldBitCap}$ be the number of remaining (or ``residual'') bits remaining after fully using $\beta_x$ field elements.`?

Reply:
Reworded (I realised I used the word "remaining" later in the sentence, so the first use was redundant).

Reply:
Should be fixed properly now with `the number of remaining or ``residual'' bits remaining after fully using $\beta_x$ field elements.` -> `the number of ``residual'' bits remaining after fully using $\beta_x$ field elements.`.
Sorry about that.

Reply:
Thanks